collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
+
+    def __get__(self, instance, owner):
```

Comment:
Ok, now I'm confused... this was bugging me before but now I think it's more obvious, so I'll address this clearly. Why are you using the __get__ and __set__ methods instead of __getattr__ and __setattr__ which look like they're a better match for what you're trying to do here? I think using the latter will be more correct and result in more readable code anyway.